### PR TITLE
{lib}[foss/2023b] HeFFTe v2.4.1 w/ CUDA 12.4.0

### PR DESCRIPTION
**Pull request by bedroge**
(created using `eb --new-pr`)

Note: the tests can only be run on a GPU node.

### CODE REVIEW

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3-a100

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
v100v2gpu1 - Linux Rocky Linux 8.10, x86_64, Intel(R) Xeon(R) Gold 6150 CPU @ 2.70GHz (skylake_avx512), 2 x NVIDIA Tesla V100-PCIE-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/bedroge/9eb95e3b48b1ea3d4bd557f42e19015f for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
l40sgpu1 - Linux Rocky Linux 8.10, x86_64, Intel(R) Xeon(R) Gold 6240R CPU @ 2.40GHz (cascadelake), 1 x NVIDIA NVIDIA L40S, 555.42.06, Python 3.6.8
See https://gist.github.com/bedroge/cd296efdde7c63a18109e87ab0f1decd for a full test report.

**Review by branfosj — DISMISSED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0208u35a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), 1 x NVIDIA NVIDIA A100-SXM4-40GB, 535.154.05, Python 3.6.8
See https://gist.github.com/branfosj/3e431b361d96ffd30f2daf4a4512df6f for a full test report.

File: easybuild/easyconfigs/h/HeFFTe/HeFFTe-2.4.1-foss-2023b-CUDA-12.4.0.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+dependencies = [
+    ('CUDA', '12.4.0', '', SYSTEM),
+    ('UCX-CUDA', '1.15.0', versionsuffix),
+]
+
+build_shared_libs = True
+
+configopts = "-DHeffte_ENABLE_FFTW=ON -DFFTW_ROOT=$EBROOTFFTW -DHeffte_ENABLE_MKL=OFF "
+configopts += "-DHeffte_ENABLE_CUDA=ON -DCUDA_TOOLKIT_ROOT_DIR=$EBROOTCUDA -DHeffte_ENABLE_GPU_AWARE_MPI=ON"
```
Comment (branfosj) [inline easybuild/easyconfigs/h/HeFFTe/HeFFTe-2.4.1-foss-2023b-CUDA-12.4.0.eb:28]:
```suggestion
configopts += "-DHeffte_ENABLE_CUDA=ON -DCUDAToolkit_ROOT=$EBROOTCUDA -DHeffte_ENABLE_GPU_AWARE_MPI=ON"
```

`CUDA_TOOLKIT_ROOT_DIR` is old CMake.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
The tests fail for me on a node with NVIDIA driver 535.154.05 and pass on a node with 560.35.05.

There are other differences between the nodes, but I suspect that that is the important difference.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0208u11a - Linux RHEL 8.10, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 560.35.05, Python 3.6.8
See https://gist.github.com/branfosj/251922ab1fe6bd6091cdc85653febaa1 for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @bedroge!

**Pull request merged**